File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/MemoryLimitConfigurationDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Configuration details for memory limits.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=MemoryLimitConfigurationDetails.Builder.class)

public final class MemoryLimitConfigurationDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"messageLimit", "tokenLimit"})
    public MemoryLimitConfigurationDetails(Integer messageLimit, Integer tokenLimit) {
        super();
        this.messageLimit = messageLimit;
        this.tokenLimit = tokenLimit;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Message-count limit for truncation middleware.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("messageLimit")
        private Integer messageLimit;

                /**
         * Message-count limit for truncation middleware.
         * @param messageLimit the value to set
         * @return this builder
         **/
        

        public Builder messageLimit(Integer messageLimit) {
        this.messageLimit = messageLimit;
        return this;
        }
            /**
     * Approximate token-count limit for truncation middleware.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tokenLimit")
        private Integer tokenLimit;

                /**
         * Approximate token-count limit for truncation middleware.
         * @param tokenLimit the value to set
         * @return this builder
         **/
        

        public Builder tokenLimit(Integer tokenLimit) {
        this.tokenLimit = tokenLimit;
        return this;
        }


        public MemoryLimitConfigurationDetails build() {
            MemoryLimitConfigurationDetails model = new MemoryLimitConfigurationDetails(this.messageLimit
                    , this.tokenLimit);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(MemoryLimitConfigurationDetails model) {
                this.messageLimit(model.getMessageLimit());
    this.tokenLimit(model.getTokenLimit());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Message-count limit for truncation middleware.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("messageLimit")
    private final Integer messageLimit;

        /**
     * Message-count limit for truncation middleware.
     * @return the value
     **/
    
    public Integer getMessageLimit() {
        return messageLimit;
    }


        /**
     * Approximate token-count limit for truncation middleware.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tokenLimit")
    private final Integer tokenLimit;

        /**
     * Approximate token-count limit for truncation middleware.
     * @return the value
     **/
    
    public Integer getTokenLimit() {
        return tokenLimit;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("MemoryLimitConfigurationDetails(");
        sb.append("messageLimit=").append(String.valueOf(this.messageLimit));
        sb.append(", tokenLimit=").append(String.valueOf(this.tokenLimit));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof MemoryLimitConfigurationDetails)) {
            return false;
        }

        MemoryLimitConfigurationDetails other = (MemoryLimitConfigurationDetails) o;
        return java.util.Objects.equals(this.messageLimit, other.messageLimit) &&
            java.util.Objects.equals(this.tokenLimit, other.tokenLimit);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.messageLimit == null ? 43 : this.messageLimit.hashCode());
        result = (result * PRIME) + (this.tokenLimit == null ? 43 : this.tokenLimit.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/memory_limit_configuration_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class MemoryLimitConfigurationDetails(object):
    """
    Configuration details for memory limits.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new MemoryLimitConfigurationDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param message_limit:
            The value to assign to the message_limit property of this MemoryLimitConfigurationDetails.
        :type message_limit: int

        :param token_limit:
            The value to assign to the token_limit property of this MemoryLimitConfigurationDetails.
        :type token_limit: int

        """
        self.swagger_types = {
            'message_limit': 'int',
            'token_limit': 'int'
        }

        self.attribute_map = {
            'message_limit': 'messageLimit',
            'token_limit': 'tokenLimit'
        }

        self._message_limit = None
        self._token_limit = None

    @property
    def message_limit(self):
        """
        Gets the message_limit of this MemoryLimitConfigurationDetails.
        Message-count limit for truncation middleware.


        :return: The message_limit of this MemoryLimitConfigurationDetails.
        :rtype: int
        """
        return self._message_limit

    @message_limit.setter
    def message_limit(self, message_limit):
        """
        Sets the message_limit of this MemoryLimitConfigurationDetails.
        Message-count limit for truncation middleware.


        :param message_limit: The message_limit of this MemoryLimitConfigurationDetails.
        :type: int
        """
        self._message_limit = message_limit

    @property
    def token_limit(self):
        """
        Gets the token_limit of this MemoryLimitConfigurationDetails.
        Approximate token-count limit for truncation middleware.


        :return: The token_limit of this MemoryLimitConfigurationDetails.
        :rtype: int
        """
        return self._token_limit

    @token_limit.setter
    def token_limit(self, token_limit):
        """
        Sets the token_limit of this MemoryLimitConfigurationDetails.
        Approximate token-count limit for truncation middleware.


        :param token_limit: The token_limit of this MemoryLimitConfigurationDetails.
        :type: int
        """
        self._token_limit = token_limit

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
